--- a/net/minecraft/server/dedicated/Settings.java
+++ b/net/minecraft/server/dedicated/Settings.java
@@ -92,13 +92,13 @@
       String s1 = this.getStringRaw(s);
       V obj1 = MoreObjects.firstNonNull((V)(s1 != null ? function.apply(s1) : null), obj);
       this.properties.put(s, function1.apply(obj1));
-      return new Settings.MutableValue<>(s, obj1, function1);
+      return new MutableValue<V>(s, obj1, function1);
    }
 
    protected <V> V get(String s, Function<String, V> function, UnaryOperator<V> unaryoperator, Function<V, String> function1, V obj) {
       return this.get(s, (sx) -> {
-         V obj = function.apply(sx);
-         return (V)(obj != null ? unaryoperator.apply(obj) : null);
+         V obj1 = function.apply(sx);
+         return (V)(obj != null ? unaryoperator.apply(obj1) : null);
       }, function1, obj);
    }
 
