--- a/net/minecraft/world/item/ItemPropertyFunction.java
+++ b/net/minecraft/world/item/ItemPropertyFunction.java
@@ -1,4 +1,10 @@
 package net.minecraft.world.item;
 
+import net.minecraft.world.entity.LivingEntity;
+import net.minecraft.world.level.Level;
+
+import javax.annotation.Nullable;
+
 public interface ItemPropertyFunction {
+    float call(ItemStack itemstack, @Nullable Level level, @Nullable LivingEntity livingentity);
 }
