--- /dev/null
+++ b/net/minecraft/world/entity/vehicle/Cannon.java
@@ -0,0 +1,84 @@
+package net.minecraft.world.entity.vehicle;
+
+import net.minecraft.world.entity.AbstractArtillery;
+import net.minecraft.world.entity.EntityType;
+import net.minecraft.world.entity.player.Player;
+import net.minecraft.world.item.Item;
+import net.minecraft.world.item.Items;
+import net.minecraft.world.level.Level;
+import net.minecraft.world.phys.Vec3;
+
+public class Cannon extends AbstractArtillery<Cannon>
+{
+    public Cannon(EntityType<? extends Cannon> entityType, Level level) {
+        super(entityType, level);
+    }
+
+    public Cannon(Level level, double x, double y, double z) {
+        super(EntityType.CANNON, level, x, y, z);
+        this.setMaxSpeed(0.00745F);
+        this.setRotationSpeed(0.355F);
+        this.setBarrelRotationSpeed(0.495F);
+    }
+
+    @Override
+    public float getBarrelAngleMin() {
+        return -17.0f;
+    }
+
+    @Override
+    public float getBarrelAngleMax() {
+        return 20.0f;
+    }
+
+    /**
+     * Returns cannon's barrel top position
+     */
+    @Override
+    protected Vec3 getBarrelTop() {
+        return this.getEyePosition(1.0F).add(this.getViewVector(1.0F).scale(1.15f));
+    }
+
+    @Override
+    protected boolean isLookingAtTrigger(Player player) {
+        Vec3 vec3dplayer = player.getViewVector(1.0F);
+        Vec3 vec3dtrigger = this.getEyePosition(1.0F).subtract(this.getViewVector(1.0F));
+        Vec3 vec3dbetween = vec3dtrigger.subtract(player.getEyePosition(1.0F)).normalize();
+
+        boolean lookingOnTrigger = vec3dbetween.dot(vec3dplayer) > 0.985D;
+        boolean behind = this.getViewVector(1.0F).multiply(1, 0, 1).dot(vec3dplayer.multiply(1, 0, 1)) > 0.0D;
+
+        return lookingOnTrigger && behind;
+    }
+
+    @Override
+    protected boolean isLookingAtBarrel(Player player) {
+        Vec3 vec3dbarrel = getBarrelTop();
+        Vec3 vec3dbetween = vec3dbarrel.subtract(player.getEyePosition(1.0F)).normalize();
+
+        Vec3 vec3dplayer = player.getViewVector(1.0F);
+        boolean lookingOnBarrel = vec3dbetween.dot(vec3dplayer) > 0.965D;
+        boolean inFront = this.getViewVector(1.0F).multiply(1, 0, 1).dot(vec3dplayer.multiply(1, 0, 1)) < 0.0D;
+
+        return lookingOnBarrel && inFront;
+    }
+
+    @Override
+    public Item getDropItem() {
+        switch (this.getWoodType()) {
+        case OAK:
+        default:
+            return Items.OAK_CANNON;
+        case SPRUCE:
+            return Items.SPRUCE_CANNON;
+        case BIRCH:
+            return Items.BIRCH_CANNON;
+        case JUNGLE:
+            return Items.JUNGLE_CANNON;
+        case ACACIA:
+            return Items.ACACIA_CANNON;
+        case DARK_OAK:
+            return Items.DARK_OAK_CANNON;
+        }
+    }
+}
