--- a/net/minecraft/client/particle/ParticleEngine.java
+++ b/net/minecraft/client/particle/ParticleEngine.java
@@ -238,7 +238,7 @@
 
    @Nullable
    private <T extends ParticleOptions> Particle makeParticle(T options, double d, double d1, double d2, double d3, double d4, double d5) {
-      ParticleProvider<T> particleprovider = this.providers.get(Registry.PARTICLE_TYPE.getId(options.getType()));
+      ParticleProvider<T> particleprovider = (ParticleProvider<T>) this.providers.get(Registry.PARTICLE_TYPE.getId(options.getType()));
       return particleprovider == null ? null : particleprovider.createParticle(options, this.level, d, d1, d2, d3, d4, d5);
    }
 
