--- a/net/minecraft/world/level/chunk/storage/IOWorker.java
+++ b/net/minecraft/world/level/chunk/storage/IOWorker.java
@@ -46,7 +46,7 @@
                if (throwable != null) {
                   completablefuture.completeExceptionally(throwable);
                } else {
-                  completablefuture.complete((Object)null);
+                  completablefuture.complete(null);
                }
 
             });
@@ -103,7 +103,7 @@
                return new CompletableFuture[i];
             }));
             completablefuture1.whenComplete((obj, throwable) -> {
-               completablefuture.complete((Object)null);
+               completablefuture.complete(null);
             });
          };
       });
