--- /dev/null
+++ b/net/minecraft/client/model/CannonModel.java
@@ -0,0 +1,111 @@
+package net.minecraft.client.model;
+
+import java.util.Arrays;
+import net.minecraft.client.model.geom.ModelPart;
+import net.minecraft.world.entity.vehicle.Cannon;
+import net.minecraftforge.api.distmarker.Dist;
+import net.minecraftforge.api.distmarker.OnlyIn;
+
+@OnlyIn(Dist.CLIENT)
+public class CannonModel extends ListModel<Cannon>
+{
+    private final ModelPart[] cubes = new ModelPart[6];
+    private ModelPart ignitor;
+
+    public CannonModel() {
+        this.cubes[0] = new ModelPart(this, 0, 10).setTexSize(128, 64); // barrel
+        this.cubes[1] = new ModelPart(this, 0, 0).setTexSize(128, 64); // frame
+        this.cubes[2] = new ModelPart(this, 15, 31).setTexSize(128, 64); // wheel 1
+        this.cubes[3] = new ModelPart(this, 15, 31).setTexSize(128, 64); // wheel 2
+        this.cubes[4] = new ModelPart(this, 15, 31).setTexSize(128, 64); // wheel 3
+        this.cubes[5] = new ModelPart(this, 15, 31).setTexSize(128, 64); // wheel 4
+
+        //Barrel with rotation
+
+        //Zapalanie
+        this.cubes[0].addBox(-2.0F, -15.0F, -2.0F, 4, 2, 4, 0.0F);
+        this.cubes[0].texOffs(2, 2).addBox(-3.5F, -13.0F, -3.5F, 7, 2, 7, 0.25F);
+
+        //Lufa 1
+        this.cubes[0].texOffs(0, 2).addBox(-3.5F, -11.0F, -3.5F, 7, 6, 7, 0.0F);
+
+        //Prążek
+        this.cubes[0].texOffs(2, 2).addBox(-3.5F, -5.0F, -3.5F, 7, 2, 7, 0.3F);
+
+        //Lufa 2
+        this.cubes[0].texOffs(0, 2).addBox(-3.5F, -3.0F, -3.5F, 7, 8, 7, 0.0F);
+
+        //Prążek
+        this.cubes[0].texOffs(0, 0).addBox(-3.5F, 5.0F, -3.5F, 7, 2, 7, 0.3F);
+
+        //Lufa 3
+        this.cubes[0].texOffs(0, 2).addBox(-3.5F, 7.0F, -3.5F, 7, 8, 7, 0.0F);
+
+        // Top lufy
+        this.cubes[0].texOffs(2, 21).addBox(-3.5F, 15.0F, -3.5F, 7, 2, 7, 0.5F);
+
+        //Zawiasy
+        this.cubes[0].addBox(-7.5F, -1.0F, -1.0F, 4, 2, 2, 0.0F);
+        this.cubes[0].addBox(3.5F, -1.0F, -1.0F, 4, 2, 2, 0.0F);
+
+        this.cubes[0].setPos(0.0F, 13.0F, 0.0F);
+        this.cubes[0].xRot = (float)-(Math.PI / 2D);
+
+        //Frame with rotation
+        this.cubes[1].texOffs(0, 31).addBox(0.0F, 0.0F, 0.0F, 23, 3, 15, 0.0F);
+        this.cubes[1].texOffs(31, 0).addBox(0.5F, 3.0F, 0.5F, 16, 9, 2, 0.0F);
+        this.cubes[1].texOffs(31, 0).addBox(0.5F, 3.0F, 12.5F, 16, 9, 2, 0.0F);
+        this.cubes[1].setPos(7.5F, 1.5F, -10.0F);
+        this.cubes[1].yRot = (float)-(Math.PI / 2D);
+
+        //Wheels with rotation
+        this.cubes[2].addBox(-2.0F, -3.0F, -1.0F, 4, 6, 2, 0.0F);
+        this.cubes[2].addBox(-3.0F, -2.0F, -1.01F, 6, 4, 2, 0.0F);
+        this.cubes[2].texOffs(0, 0).addBox(-1.0F, -1.0F, -1.5F, 2, 2, 3, -0.1F);
+        this.cubes[2].yRot = (float)-(Math.PI / 2);
+
+        this.cubes[3].addBox(-2.0F, -3.0F, -1.0F, 4, 6, 2, 0.0F);
+        this.cubes[3].addBox(-3.0F, -2.0F, -1.01F, 6, 4, 2, 0.0F);
+        this.cubes[3].texOffs(0, 0).addBox(-1.0F, -1.0F, -1.5F, 2, 2, 3, -0.1F);
+        this.cubes[3].yRot = (float)-(Math.PI / 2);
+
+        this.cubes[4].addBox(-2.0F, -3.0F, -1.0F, 4, 6, 2, 0.0F);
+        this.cubes[4].addBox(-3.0F, -2.0F, -1.01F, 6, 4, 2, 0.0F);
+        this.cubes[4].texOffs(0, 0).addBox(-1.0F, -1.0F, -1.5F, 2, 2, 3, -0.1F);
+        this.cubes[4].yRot = (float)-(Math.PI / 2);
+
+        this.cubes[5].addBox(-2.0F, -3.0F, -1.0F, 4, 6, 2, 0.0F);
+        this.cubes[5].addBox(-3.0F, -2.0F, -1.01F, 6, 4, 2, 0.0F);
+        this.cubes[5].texOffs(0, 0).addBox(-1.0F, -1.0F, -1.5F, 2, 2, 3, -0.1F);
+        this.cubes[5].yRot = (float)-(Math.PI / 2);
+
+        //Wheels positioning
+        this.cubes[2].setPos(-8.75F, 3.0F, 11.0F);
+        this.cubes[3].setPos(8.75F, 3.0F, 11.0F);
+        this.cubes[4].setPos(-8.75F, 3.0F, -8.0F);
+        this.cubes[5].setPos(8.75F, 3.0F, -8.0F);
+        
+        // Add ignitor
+        this.ignitor = new ModelPart(this).setTexSize(128, 64);
+        ignitor.texOffs(0, 31).addBox(-1.0F, -1.0F, 0.0F, 2, 3, 2, -0.5F);
+        ignitor.setPos(0.0F, -13.0F, 2.0F);
+        ignitor.xRot = 1.85F;
+        this.cubes[0].addChild(ignitor);
+    }
+
+    /**
+     * Sets this entity's model rotation angles
+     */
+    public void setupAnim(Cannon p_225597_1_, float p_225597_2_, float p_225597_3_, float p_225597_4_,
+        float p_225597_5_, float p_225597_6_) {
+        // Rotate barrel
+        this.cubes[0].xRot = (float)(-(p_225597_1_.xRot * (Math.PI / 180F)) - (Math.PI / 2));
+        
+        // Show ignitor when cannon ready to shoot
+        this.ignitor.visible = p_225597_1_.isReady();
+    }
+
+    public Iterable<ModelPart> parts() {
+        return Arrays.asList(this.cubes);
+    }
+}
