--- a/net/minecraft/util/profiling/FilledProfileResults.java
+++ b/net/minecraft/util/profiling/FilledProfileResults.java
@@ -123,7 +123,7 @@
          if (!map1.isEmpty()) {
             List<String> list = SPLITTER.splitToList(s);
             map1.forEach((sx, i) -> {
-               map.computeIfAbsent(sx, (s) -> {
+               map.computeIfAbsent(sx, (s1) -> {
                   return new FilledProfileResults.CounterCollector();
                }).addValue(list.iterator(), i);
             });
