--- a/net/minecraft/advancements/critereon/PlayerPredicate.java
+++ b/net/minecraft/advancements/critereon/PlayerPredicate.java
@@ -181,11 +181,11 @@
          if (!this.stats.isEmpty()) {
             JsonArray array = new JsonArray();
             this.stats.forEach((stat, ints) -> {
-               JsonObject jsonobject = new JsonObject();
-               jsonobject.addProperty("type", Registry.STAT_TYPE.getKey(stat.getType()).toString());
-               jsonobject.addProperty("stat", getStatValueId(stat).toString());
-               jsonobject.add("value", ints.serializeToJson());
-               array.add(jsonobject);
+               JsonObject jsonobject1 = new JsonObject();
+               jsonobject1.addProperty("type", Registry.STAT_TYPE.getKey(stat.getType()).toString());
+               jsonobject1.addProperty("stat", getStatValueId(stat).toString());
+               jsonobject1.add("value", ints.serializeToJson());
+               array.add(jsonobject1);
             });
             jsonobject.add("stats", array);
          }
