--- a/net/minecraft/util/datafix/fixes/VillagerRebuildLevelAndXpFix.java
+++ b/net/minecraft/util/datafix/fixes/VillagerRebuildLevelAndXpFix.java
@@ -37,10 +37,10 @@
             int i = dynamic.get("VillagerData").get("level").asNumber().orElse(0).intValue();
             Typed<?> typed1 = typedx;
             if (i == 0 || i == 1) {
-               int i1 = typedx.getOptionalTyped(opticfinder1).flatMap((typed) -> {
-                  return typed.getOptionalTyped(opticfinder2);
-               }).map((typed) -> {
-                  return typed.getAllTyped(opticfinder3).size();
+               int i1 = typedx.getOptionalTyped(opticfinder1).flatMap((typed2) -> {
+                  return typed2.getOptionalTyped(opticfinder2);
+               }).map((typed3) -> {
+                  return typed3.getAllTyped(opticfinder3).size();
                }).orElse(0);
                i = Mth.clamp(i1 / 2, 1, 5);
                if (i > 1) {
