--- a/net/minecraft/util/datafix/fixes/OptionsKeyLwjgl3Fix.java
+++ b/net/minecraft/util/datafix/fixes/OptionsKeyLwjgl3Fix.java
@@ -1,9 +1,6 @@
 package net.minecraft.util.datafix.fixes;
 
-import com.mojang.datafixers.DSL;
-import com.mojang.datafixers.DataFix;
-import com.mojang.datafixers.DataFixUtils;
-import com.mojang.datafixers.TypeRewriteRule;
+import com.mojang.datafixers.*;
 import com.mojang.datafixers.schemas.Schema;
 import com.mojang.datafixers.util.Pair;
 import it.unimi.dsi.fastutil.ints.Int2ObjectMap;
@@ -159,7 +156,7 @@
                      return Pair.of(entry.getKey(), entry.getValue());
                   }
                }).collect(Collectors.toMap(Pair::getFirst, Pair::getSecond)));
-            }).orElse(dynamic);
+            }).orElse((Dynamic) dynamic);
          });
       });
    }
