--- a/net/minecraft/world/level/timers/TimerQueue.java
+++ b/net/minecraft/world/level/timers/TimerQueue.java
@@ -23,7 +23,7 @@
    private final Table<String, Long, TimerQueue.Event<T>> events = HashBasedTable.create();
 
    private static <T> Comparator<TimerQueue.Event<T>> createComparator() {
-      return Comparator.comparingLong((event) -> {
+      return Comparator.<TimerQueue.Event<T>>comparingLong((event) -> {
          return event.triggerTime;
       }).thenComparing((event) -> {
          return event.sequentialId;
