--- a/net/minecraft/advancements/CriteriaTriggers.java
+++ b/net/minecraft/advancements/CriteriaTriggers.java
@@ -91,7 +91,7 @@
 
    @Nullable
    public static <T extends CriterionTriggerInstance> CriterionTrigger<T> getCriterion(ResourceLocation resource) {
-      return CRITERIA.get(resource);
+      return (CriterionTrigger<T>) CRITERIA.get(resource);
    }
 
    public static Iterable<? extends CriterionTrigger<?>> all() {
