--- a/net/minecraft/server/commands/ForceLoadCommand.java
+++ b/net/minecraft/server/commands/ForceLoadCommand.java
@@ -16,6 +16,8 @@
 import net.minecraft.world.level.ChunkPos;
 import net.minecraft.world.level.dimension.DimensionType;
 
+import java.util.function.LongConsumer;
+
 public class ForceLoadCommand {
    private static final Dynamic2CommandExceptionType ERROR_TOO_MANY_CHUNKS = new Dynamic2CommandExceptionType((obj, obj1) -> {
       return new TranslatableComponent("commands.forceload.toobig", obj, obj1);
@@ -80,7 +82,7 @@
       DimensionType type = commandsourcestack.getLevel().getDimension().getType();
       ServerLevel level = commandsourcestack.getServer().getLevel(type);
       LongSet set = level.getForcedChunks();
-      set.forEach((i) -> {
+      set.forEach((long i) -> {
          level.setChunkForced(ChunkPos.getX(i), ChunkPos.getZ(i), false);
       });
       commandsourcestack.sendSuccess(new TranslatableComponent("commands.forceload.removed.all", type), true);
