--- a/net/minecraft/util/datafix/fixes/BedBlockEntityInjecter.java
+++ b/net/minecraft/util/datafix/fixes/BedBlockEntityInjecter.java
@@ -37,10 +37,10 @@
       Type<TE> type2 = type1.getElement();
       OpticFinder<?> opticfinder = DSL.fieldFinder("Level", type);
       OpticFinder<List<TE>> opticfinder1 = DSL.fieldFinder("TileEntities", type1);
-      int i = 416;
+
       return TypeRewriteRule.seq(this.fixTypeEverywhere("InjectBedBlockEntityType", this.getInputSchema().findChoiceType(References.BLOCK_ENTITY), this.getOutputSchema().findChoiceType(References.BLOCK_ENTITY), (dynamicops) -> {
          return (pair) -> {
-            return pair;
+            return (com.mojang.datafixers.util.Pair) pair;
          };
       }), this.fixTypeEverywhereTyped("BedBlockEntityInjecter", this.getOutputSchema().getType(References.CHUNK), (typed) -> {
          Typed<?> typed1 = typed.getTyped(opticfinder);
@@ -58,7 +58,7 @@
             });
             int i4 = 0;
 
-            for(int i5 : stream::iterator) {
+            for(int i5 : (Iterable<Integer>) stream::iterator) {
                if (416 == (i5 & 255) << 4) {
                   int i6 = i4 & 15;
                   int i7 = i4 >> 8 & 15;
