--- a/net/minecraft/util/datafix/fixes/EntityIdFix.java
+++ b/net/minecraft/util/datafix/fixes/EntityIdFix.java
@@ -17,6 +17,8 @@
       map.put("Bat", "minecraft:bat");
       map.put("Blaze", "minecraft:blaze");
       map.put("Boat", "minecraft:boat");
+      map.put("Cannon", "minecraft:cannon");
+      map.put("Cannonball", "minecraft:cannonball");
       map.put("CaveSpider", "minecraft:cave_spider");
       map.put("Chicken", "minecraft:chicken");
       map.put("Cow", "minecraft:cow");
@@ -93,8 +95,8 @@
    }
 
    public TypeRewriteRule makeRule() {
-      TaggedChoiceType<String> type = this.getInputSchema().findChoiceType(References.ENTITY);
-      TaggedChoiceType<String> type1 = this.getOutputSchema().findChoiceType(References.ENTITY);
+      TaggedChoiceType<String> type = (TaggedChoiceType<String>) this.getInputSchema().findChoiceType(References.ENTITY);
+      TaggedChoiceType<String> type1 = (TaggedChoiceType<String>) this.getOutputSchema().findChoiceType(References.ENTITY);
       Type<?> type2 = this.getInputSchema().getType(References.ITEM_STACK);
       Type<?> type3 = this.getOutputSchema().getType(References.ITEM_STACK);
       return TypeRewriteRule.seq(this.convertUnchecked("item stack entity name hook converter", type2, type3), this.fixTypeEverywhere("EntityIdFix", type, type1, (dynamicops) -> {
