--- a/net/minecraft/util/datafix/fixes/ItemLoreFix.java
+++ b/net/minecraft/util/datafix/fixes/ItemLoreFix.java
@@ -24,8 +24,8 @@
          return typed.updateTyped(opticfinder, (typedx) -> {
             return typedx.update(DSL.remainderFinder(), (dynamic) -> {
                return dynamic.update("display", (dynamicx) -> {
-                  return dynamicx.update("Lore", (dynamic) -> {
-                     return DataFixUtils.orElse(dynamic.asStreamOpt().map(ItemLoreFix::fixLoreList).map(dynamic::createList), dynamic);
+                  return dynamicx.update("Lore", (dynamic2) -> {
+                     return DataFixUtils.orElse(dynamic2.asStreamOpt().map(ItemLoreFix::fixLoreList).map(dynamic2::createList), dynamic2);
                   });
                });
             });
