--- a/net/minecraft/util/datafix/fixes/SavedDataVillageCropFix.java
+++ b/net/minecraft/util/datafix/fixes/SavedDataVillageCropFix.java
@@ -24,7 +24,7 @@
    }
 
    private static Stream<? extends Dynamic<?>> updateChildren(Stream<? extends Dynamic<?>> stream) {
-      return stream.map((dynamic) -> {
+      return stream.map((Dynamic<?> dynamic) -> {
          String s = dynamic.get("id").asString("");
          if ("ViF".equals(s)) {
             return updateSingleField(dynamic);
