--- a/net/minecraft/world/level/TickNextTickData.java
+++ b/net/minecraft/world/level/TickNextTickData.java
@@ -37,7 +37,7 @@
    }
 
    public static <T> Comparator<TickNextTickData<T>> createTimeComparator() {
-      return Comparator.comparingLong((ticknexttickdata) -> {
+      return Comparator.<TickNextTickData<T>>comparingLong((ticknexttickdata) -> {
          return ticknexttickdata.delay;
       }).thenComparing((ticknexttickdata) -> {
          return ticknexttickdata.priority;
