--- a/net/minecraft/sounds/SoundEvents.java
+++ b/net/minecraft/sounds/SoundEvents.java
@@ -34,6 +34,9 @@
    public static final SoundEvent ARROW_HIT = register("entity.arrow.hit");
    public static final SoundEvent ARROW_HIT_PLAYER = register("entity.arrow.hit_player");
    public static final SoundEvent ARROW_SHOOT = register("entity.arrow.shoot");
+    public static final SoundEvent ARTILLERY_SHOOT = register("entity.artillery.shoot");
+    public static final SoundEvent ARTILLERY_LOAD_EXPLOSIVE = register("entity.artillery.load.explosive");
+    public static final SoundEvent ARTILLERY_LOAD_BULLET = register("entity.artillery.load.bullet");
    public static final SoundEvent AXE_STRIP = register("item.axe.strip");
    public static final SoundEvent BAMBOO_BREAK = register("block.bamboo.break");
    public static final SoundEvent BAMBOO_FALL = register("block.bamboo.fall");
