--- a/com/mojang/realmsclient/gui/screens/RealmsSelectWorldTemplateScreen.java
+++ b/com/mojang/realmsclient/gui/screens/RealmsSelectWorldTemplateScreen.java
@@ -194,10 +194,10 @@
    private void fetchTemplatesAsync(final WorldTemplatePaginatedList list) {
       (new Thread("realms-template-fetcher") {
          public void run() {
-            WorldTemplatePaginatedList list = list;
-
-            Either<WorldTemplatePaginatedList, String> either;
-            for(RealmsClient realmsclient = RealmsClient.createRealmsClient(); list != null; list = Realms.execute(() -> {
+            RealmsClient realmsclient = RealmsClient.createRealmsClient();
+            WorldTemplatePaginatedList finalList = list;
+            for(WorldTemplatePaginatedList list1 = list; list1 != null; list1 = Realms.execute(() -> {
+               Either<WorldTemplatePaginatedList, String> either = RealmsSelectWorldTemplateScreen.this.fetchTemplates(list, realmsclient);
                if (either.right().isPresent()) {
                   RealmsSelectWorldTemplateScreen.LOGGER.error("Couldn't fetch templates: {}", either.right().get());
                   if (RealmsSelectWorldTemplateScreen.this.worldTemplateObjectSelectionList.isEmpty()) {
@@ -208,13 +208,13 @@
                } else {
                   assert either.left().isPresent();
 
-                  WorldTemplatePaginatedList list = either.left().get();
+                  WorldTemplatePaginatedList list2 = either.left().get();
 
-                  for(WorldTemplate worldtemplate : list.templates) {
+                  for(WorldTemplate worldtemplate : list2.templates) {
                      RealmsSelectWorldTemplateScreen.this.worldTemplateObjectSelectionList.addEntry(worldtemplate);
                   }
 
-                  if (list.templates.isEmpty()) {
+                  if (list2.templates.isEmpty()) {
                      if (RealmsSelectWorldTemplateScreen.this.worldTemplateObjectSelectionList.isEmpty()) {
                         String s = RealmsScreen.getLocalizedString("mco.template.select.none", "%link");
                         TextRenderingUtils.LineSegment linesegment = TextRenderingUtils.LineSegment.link(RealmsScreen.getLocalizedString("mco.template.select.none.linkTitle"), "https://minecraft.net/realms/content-creator/");
@@ -223,14 +223,11 @@
 
                      return null;
                   } else {
-                     return list;
+                     return list2;
                   }
                }
-            }).join()) {
-               either = RealmsSelectWorldTemplateScreen.this.fetchTemplates(list, realmsclient);
+            }).join());
             }
-
-         }
       }).start();
    }
 
