--- a/net/minecraft/client/renderer/texture/Stitcher.java
+++ b/net/minecraft/client/renderer/texture/Stitcher.java
@@ -13,7 +13,7 @@
 
 @OnlyIn(Dist.CLIENT)
 public class Stitcher {
-   private static final Comparator<Stitcher.Holder> HOLDER_COMPARATOR = Comparator.comparing((holder) -> {
+   private static final Comparator<Stitcher.Holder> HOLDER_COMPARATOR = Comparator.<Stitcher.Holder, Integer>comparing((holder) -> {
       return -holder.height;
    }).thenComparing((holder) -> {
       return -holder.width;
