--- a/net/minecraft/client/renderer/entity/EntityRenderDispatcher.java
+++ b/net/minecraft/client/renderer/entity/EntityRenderDispatcher.java
@@ -194,9 +194,9 @@
       if (entity instanceof AbstractClientPlayer) {
          String s = ((AbstractClientPlayer)entity).getModelName();
          PlayerRenderer playerrenderer = this.playerRenderers.get(s);
-         return playerrenderer != null ? playerrenderer : this.defaultPlayerRenderer;
+         return (EntityRenderer<? super T>) (playerrenderer != null ? playerrenderer : this.defaultPlayerRenderer);
       } else {
-         return this.renderers.get(entity.getType());
+         return (EntityRenderer<? super T>) this.renderers.get(entity.getType());
       }
    }
 
