--- a/com/mojang/blaze3d/font/TrueTypeGlyphProvider.java
+++ b/com/mojang/blaze3d/font/TrueTypeGlyphProvider.java
@@ -50,7 +50,6 @@
       if (this.skip.contains(c)) {
          return null;
       } else {
-         Object i1;
          try (MemoryStack memorystack = MemoryStack.stackPush()) {
             IntBuffer buffer = memorystack.mallocInt(1);
             IntBuffer buffer1 = memorystack.mallocInt(1);
@@ -70,11 +69,9 @@
 
                return null;
             }
-
-            i1 = null;
          }
 
-         return (TrueTypeGlyphProvider.Glyph)i1;
+         return null;
       }
    }
 
