--- a/net/minecraft/world/inventory/MenuType.java
+++ b/net/minecraft/world/inventory/MenuType.java
@@ -1,6 +1,7 @@
 package net.minecraft.world.inventory;
 
 import net.minecraft.core.Registry;
+import net.minecraft.world.entity.player.Inventory;
 
 public class MenuType<T extends AbstractContainerMenu> {
    public static final MenuType<ChestMenu> GENERIC_9x1 = register("generic_9x1", ChestMenu::oneRow);
@@ -39,5 +40,6 @@
    }
 
    interface MenuSupplier<T extends AbstractContainerMenu> {
+      T call(int p_39995_, Inventory p_39996_);
    }
 }
