--- a/net/minecraft/world/level/chunk/storage/ChunkSerializer.java
+++ b/net/minecraft/world/level/chunk/storage/ChunkSerializer.java
@@ -242,8 +242,8 @@
 
       for(int i = -1; i < 17; ++i) {
          int i1 = i;
-         LevelChunkSection levelchunksection = Arrays.stream(alevelchunksection).filter((levelchunksection) -> {
-            return levelchunksection != null && levelchunksection.bottomBlockY() >> 4 == i1;
+         LevelChunkSection levelchunksection = Arrays.stream(alevelchunksection).filter((levelchunksection1) -> {
+            return levelchunksection1 != null && levelchunksection1.bottomBlockY() >> 4 == i1;
          }).findFirst().orElse(LevelChunk.EMPTY_SECTION);
          DataLayer datalayer = levellightengine.getLayerListener(LightLayer.BLOCK).getDataLayerData(SectionPos.of(chunkpos, i1));
          DataLayer datalayer1 = levellightengine.getLayerListener(LightLayer.SKY).getDataLayerData(SectionPos.of(chunkpos, i1));
