--- a/net/minecraft/world/entity/ai/behavior/InteractWith.java
+++ b/net/minecraft/world/entity/ai/behavior/InteractWith.java
@@ -50,7 +50,7 @@
          list.stream().filter((livingentityx) -> {
             return this.type.equals(livingentityx.getType());
          }).map((livingentityx) -> {
-            return livingentityx;
+            return (T) livingentityx;
          }).filter((livingentity1) -> {
             return livingentity1.distanceToSqr(livingentity) <= (double)this.interactionRangeSqr;
          }).filter(this.targetFilter).findFirst().ifPresent((livingentityx) -> {
