--- a/net/minecraft/world/entity/ai/gossip/GossipContainer.java
+++ b/net/minecraft/world/entity/ai/gossip/GossipContainer.java
@@ -81,9 +81,9 @@
    public void transferFrom(GossipContainer gossipcontainer, Random random, int i) {
       Collection<GossipContainer.GossipEntry> collection = gossipcontainer.selectGossipsForTransfer(random, i);
       collection.forEach((gossipentry) -> {
-         int i = gossipentry.value - gossipentry.type.decayPerTransfer;
-         if (i >= 2) {
-            this.getOrCreate(gossipentry.target).entries.mergeInt(gossipentry.type, i, GossipContainer::mergeValuesForTransfer);
+         int i1 = gossipentry.value - gossipentry.type.decayPerTransfer;
+         if (i1 >= 2) {
+            this.getOrCreate(gossipentry.target).entries.mergeInt(gossipentry.type, i1, GossipContainer::mergeValuesForTransfer);
          }
 
       });
