--- /dev/null
+++ b/net/minecraft/world/item/CannonballItem.java
@@ -0,0 +1,23 @@
+package net.minecraft.world.item;
+
+import net.minecraft.world.entity.projectile.Cannonball;
+import net.minecraft.world.entity.projectile.Cannonball.Type;
+
+public class CannonballItem extends Item
+{
+    private final Cannonball.Type type;
+
+    public CannonballItem(Type type, Item.Properties properties) {
+        super(properties);
+        this.type = type;
+    }
+
+    public Cannonball.Type getType() {
+        return this.type;
+    }
+
+//    public Cannonball.Type getType(ItemStack itemstack) {
+//        CompoundTag compoundnbt = itemstack.getTag();
+//        return compoundnbt != null ? Cannonball.Type.values()[compoundnbt.getByte("Type")] : Cannonball.Type.IMPACT;
+//    }
+}
