--- a/net/minecraft/world/item/Items.java
+++ b/net/minecraft/world/item/Items.java
@@ -898,6 +898,8 @@
    public static final Item HONEY_BOTTLE = registerItem("honey_bottle", new HoneyBottleItem((new Item.Properties()).craftRemainder(GLASS_BOTTLE).food(Foods.HONEY_BOTTLE).tab(CreativeModeTab.TAB_FOOD).stacksTo(16)));
    public static final Item HONEY_BLOCK = registerBlock(Blocks.HONEY_BLOCK, CreativeModeTab.TAB_DECORATIONS);
    public static final Item HONEYCOMB_BLOCK = registerBlock(Blocks.HONEYCOMB_BLOCK, CreativeModeTab.TAB_DECORATIONS);
+   public static final Item CANNON = registerItem("cannon", new CannonItem((new Item.Properties()).stacksTo(1).tab(CreativeModeTab.TAB_TRANSPORTATION)));
+   public static final Item CANNONBALL = registerItem("cannonball", new CannonballItem((new Item.Properties()).stacksTo(8).tab(CreativeModeTab.TAB_TRANSPORTATION)));
 
    private static Item registerBlock(Block block) {
       return registerBlock(new BlockItem(block, new Item.Properties()));
