--- a/net/minecraft/util/datafix/schemas/V99.java
+++ b/net/minecraft/util/datafix/schemas/V99.java
@@ -270,10 +270,10 @@
    protected static <T> T addNames(Dynamic<T> dynamic, Map<String, String> map, String s) {
       return dynamic.update("tag", (dynamic1) -> {
          return dynamic1.update("BlockEntityTag", (dynamic1x) -> {
-            String s = dynamic.get("id").asString("");
-            String s1 = map.get(NamespacedSchema.ensureNamespaced(s));
+            String id = dynamic.get("id").asString("");
+            String s1 = map.get(NamespacedSchema.ensureNamespaced(id));
             if (s1 == null) {
-               LOGGER.warn("Unable to resolve BlockEntity for ItemStack: {}", (Object)s);
+               LOGGER.warn("Unable to resolve BlockEntity for ItemStack: {}", (Object)id);
                return dynamic1x;
             } else {
                return dynamic1x.set("id", dynamic.createString(s1));
