--- a/net/minecraft/world/level/storage/LevelData.java
+++ b/net/minecraft/world/level/storage/LevelData.java
@@ -286,7 +286,7 @@
       this.generateMapFeatures = levelsettings.isGenerateMapFeatures();
       this.hardcore = levelsettings.isHardcore();
       this.generator = levelsettings.getLevelType();
-      this.setGeneratorOptions(Dynamic.convert(JsonOps.INSTANCE, NbtOps.INSTANCE, levelsettings.getLevelTypeOptions()));
+      this.setGeneratorOptions((CompoundTag) Dynamic.convert(JsonOps.INSTANCE, NbtOps.INSTANCE, levelsettings.getLevelTypeOptions()));
       this.allowCommands = levelsettings.getAllowCommands();
    }
 
