--- a/net/minecraft/client/gui/screens/MenuScreens.java
+++ b/net/minecraft/client/gui/screens/MenuScreens.java
@@ -8,6 +8,7 @@
 import net.minecraft.client.gui.screens.inventory.BeaconScreen;
 import net.minecraft.client.gui.screens.inventory.BlastFurnaceScreen;
 import net.minecraft.client.gui.screens.inventory.BrewingStandScreen;
+import net.minecraft.client.gui.screens.inventory.ArtilleryScreen;
 import net.minecraft.client.gui.screens.inventory.CartographyTableScreen;
 import net.minecraft.client.gui.screens.inventory.ContainerScreen;
 import net.minecraft.client.gui.screens.inventory.CraftingScreen;
@@ -53,7 +54,7 @@
 
    @Nullable
    private static <T extends AbstractContainerMenu> MenuScreens.ScreenConstructor<T, ?> getConstructor(MenuType<T> type) {
-      return SCREENS.get(type);
+      return (ScreenConstructor<T, ?>) SCREENS.get(type);
    }
 
    private static <M extends AbstractContainerMenu, U extends Screen & MenuAccess<M>> void register(MenuType<? extends M> type, MenuScreens.ScreenConstructor<M, U> screenconstructor) {
@@ -100,6 +101,7 @@
       register(MenuType.SMOKER, SmokerScreen::new);
       register(MenuType.CARTOGRAPHY_TABLE, CartographyTableScreen::new);
       register(MenuType.STONECUTTER, StonecutterScreen::new);
+        register(MenuType.ARTILLERY, ArtilleryScreen::new);
    }
 
    @OnlyIn(Dist.CLIENT)
