--- a/net/minecraft/client/Minecraft.java
+++ b/net/minecraft/client/Minecraft.java
@@ -471,7 +471,7 @@
    }
 
    private String createTitle() {
-      StringBuilder builder = new StringBuilder("Minecraft");
+      StringBuilder builder = new StringBuilder("MinecraftPlus");
       if (this.isProbablyModded()) {
          builder.append("*");
       }
@@ -496,7 +496,7 @@
    }
 
    public boolean isProbablyModded() {
-      return !"vanilla".equals(ClientBrandRetriever.getClientModName()) || Minecraft.class.getSigners() == null;
+      return !"minecraftplus".equals(ClientBrandRetriever.getClientModName()) || Minecraft.class.getSigners() == null;
    }
 
    private void rollbackResourcePacks(Throwable throwable) {
@@ -1807,9 +1807,9 @@
          return "Yes";
       });
       category.setDetail("Is Modded", () -> {
-         String s = ClientBrandRetriever.getClientModName();
-         if (!"vanilla".equals(s)) {
-            return "Definitely; Client brand changed to '" + s + "'";
+         String brand = ClientBrandRetriever.getClientModName();
+         if (!"vanilla".equals(brand)) {
+            return "Definitely; Client brand changed to '" + brand + "'";
          } else {
             return Minecraft.class.getSigners() == null ? "Very likely; Jar signature invalidated" : "Probably not. Jar signature remains and client brand is untouched.";
          }
@@ -1819,13 +1819,13 @@
          category.setDetail("Resource Packs", () -> {
             StringBuilder builder = new StringBuilder();
 
-            for(String s : options.resourcePacks) {
+            for(String pack : options.resourcePacks) {
                if (builder.length() > 0) {
                   builder.append(", ");
                }
 
-               builder.append(s);
-               if (options.incompatibleResourcePacks.contains(s)) {
+               builder.append(pack);
+               if (options.incompatibleResourcePacks.contains(pack)) {
                   builder.append(" (incompatible)");
                }
             }
