--- a/net/minecraft/world/entity/boss/enderdragon/EndCrystal.java
+++ b/net/minecraft/world/entity/boss/enderdragon/EndCrystal.java
@@ -14,6 +14,7 @@
 import net.minecraft.world.entity.Entity;
 import net.minecraft.world.entity.EntityType;
 import net.minecraft.world.level.Explosion;
+import net.minecraft.world.level.GameRules;
 import net.minecraft.world.level.Level;
 import net.minecraft.world.level.block.Blocks;
 import net.minecraft.world.level.dimension.end.EndDragonFight;
@@ -89,7 +90,8 @@
          if (!this.removed && !this.level.isClientSide) {
             this.remove();
             if (!damagesource.isExplosion()) {
-               this.level.explode((Entity)null, this.getX(), this.getY(), this.getZ(), 6.0F, Explosion.BlockInteraction.DESTROY);
+               boolean ruleDoExplosions = this.level.getGameRules().getBoolean(GameRules.RULE_DO_EXPLOSIONS);
+               this.level.explode((Entity)null, this.getX(), this.getY(), this.getZ(), 6.0F, ruleDoExplosions ? Explosion.BlockInteraction.BREAK : Explosion.BlockInteraction.NONE);
             }
 
             this.onDestroyedBy(damagesource);
