--- a/net/minecraft/client/player/LocalPlayer.java
+++ b/net/minecraft/client/player/LocalPlayer.java
@@ -58,6 +58,7 @@
 import net.minecraft.world.entity.Pose;
 import net.minecraft.world.entity.vehicle.AbstractMinecart;
 import net.minecraft.world.entity.vehicle.Boat;
+import net.minecraft.world.entity.vehicle.Cannon;
 import net.minecraft.world.item.ElytraItem;
 import net.minecraft.world.item.Item;
 import net.minecraft.world.item.ItemStack;
@@ -153,6 +154,12 @@
             this.setYHeadRot(entity.yRot);
          }
 
+         if (entity instanceof Cannon) {
+            this.yRotO = entity.yRot;
+            this.yRot = entity.yRot;
+            this.setYHeadRot(entity.yRot);
+         }
+
          return true;
       }
    }
@@ -781,6 +788,10 @@
          Boat boat = (Boat)this.getVehicle();
          boat.setInput(this.input.left, this.input.right, this.input.up, this.input.down);
          this.handsBusy |= this.input.left || this.input.right || this.input.up || this.input.down;
+      } else if (this.getVehicle() instanceof Cannon) {
+         Cannon cannon = (Cannon)this.getVehicle();
+         cannon.setInput(this.input.left, this.input.right, this.input.up, this.input.down, this.input.jumping);
+         this.handsBusy |= this.input.left || this.input.right || this.input.up || this.input.down || this.input.jumping;
       }
 
    }
