--- a/net/minecraft/server/packs/resources/ProfiledReloadInstance.java
+++ b/net/minecraft/server/packs/resources/ProfiledReloadInstance.java
@@ -27,7 +27,7 @@
          ActiveProfiler activeprofiler1 = new ActiveProfiler(Util.getNanos(), () -> {
             return 0;
          }, false);
-         CompletableFuture<Void> completablefuture = preparablereloadlistener.reload(preparationbarrier, managerx, activeprofiler, activeprofiler1, (runnable) -> {
+         CompletableFuture<Void> completablefuture1 = preparablereloadlistener.reload(preparationbarrier, managerx, activeprofiler, activeprofiler1, (runnable) -> {
             executor1x.execute(() -> {
                long i = Util.getNanos();
                runnable.run();
@@ -40,7 +40,7 @@
                atomiclong1.addAndGet(Util.getNanos() - i);
             });
          });
-         return completablefuture.thenApplyAsync((nothing) -> {
+         return completablefuture1.thenApplyAsync((nothing) -> {
             return new ProfiledReloadInstance.State(preparablereloadlistener.getName(), activeprofiler.getResults(), activeprofiler1.getResults(), atomiclong, atomiclong1);
          }, executor1);
       }, completablefuture);
