--- a/net/minecraft/commands/synchronization/ArgumentTypes.java
+++ b/net/minecraft/commands/synchronization/ArgumentTypes.java
@@ -130,7 +130,7 @@
    }
 
    public static <T extends ArgumentType<?>> void serialize(FriendlyByteBuf buffer, T type) {
-      ArgumentTypes.Entry<T> entry = get(type);
+      ArgumentTypes.Entry<T> entry = (Entry<T>) get(type);
       if (entry == null) {
          LOGGER.error("Could not serialize {} ({}) - will not be sent to client!", type, type.getClass());
          buffer.writeResourceLocation(new ResourceLocation(""));
@@ -153,7 +153,7 @@
    }
 
    private static <T extends ArgumentType<?>> void serializeToJson(JsonObject jsonobject, T type) {
-      ArgumentTypes.Entry<T> entry = get(type);
+      ArgumentTypes.Entry<T> entry = (Entry<T>) get(type);
       if (entry == null) {
          LOGGER.error("Could not serialize argument {} ({})!", type, type.getClass());
          jsonobject.addProperty("type", "unknown");
