--- a/net/minecraft/data/tags/ItemTagsProvider.java
+++ b/net/minecraft/data/tags/ItemTagsProvider.java
@@ -84,7 +84,7 @@
       } else if (entry instanceof Tag.ValuesEntry) {
          List<Item> list = Lists.newArrayList();
 
-         for(Block block : ((Tag.ValuesEntry)entry).getValues()) {
+         for(Block block : ((Tag.ValuesEntry<Block>)entry).getValues()) {
             Item item = block.asItem();
             if (item == Items.AIR) {
                LOGGER.warn("Itemless block copied to item tag: {}", (Object)Registry.BLOCK.getKey(block));
