--- a/net/minecraft/client/renderer/texture/HttpTexture.java
+++ b/net/minecraft/client/renderer/texture/HttpTexture.java
@@ -107,9 +107,9 @@
                      }
 
                      Minecraft.getInstance().execute(() -> {
-                        NativeImage nativeimage = this.load(stream);
-                        if (nativeimage != null) {
-                           this.loadCallback(nativeimage);
+                        NativeImage nativeimage1 = this.load(stream);
+                        if (nativeimage1 != null) {
+                           this.loadCallback(nativeimage1);
                         }
 
                      });
