--- a/net/minecraft/world/level/storage/loot/entries/EntryGroup.java
+++ b/net/minecraft/world/level/storage/loot/entries/EntryGroup.java
@@ -23,8 +23,8 @@
          };
       default:
          return (context, consumer) -> {
-            for(ComposableEntryContainer composableentrycontainer : acomposableentrycontainer) {
-               composableentrycontainer.expand(context, consumer);
+            for(ComposableEntryContainer container : acomposableentrycontainer) {
+               container.expand(context, consumer);
             }
 
             return true;
