--- a/net/minecraft/world/level/storage/loot/entries/LootPoolEntries.java
+++ b/net/minecraft/world/level/storage/loot/entries/LootPoolEntries.java
@@ -65,7 +65,7 @@
          if (serializer == null) {
             throw new JsonParseException("Unknown item type: " + clazz);
          } else {
-            return serializer;
+            return (LootPoolEntryContainer.Serializer<LootPoolEntryContainer>) serializer;
          }
       }
    }
