--- a/net/minecraft/world/level/levelgen/feature/treedecorators/AlterGroundDecorator.java
+++ b/net/minecraft/world/level/levelgen/feature/treedecorators/AlterGroundDecorator.java
@@ -37,7 +37,7 @@
          this.placeCircle(levelaccessor, random, blockpos.west().south(2));
          this.placeCircle(levelaccessor, random, blockpos.east(2).south(2));
 
-         for(int i = 0; i < 5; ++i) {
+         for(int j = 0; j < 5; ++j) {
             int i1 = random.nextInt(64);
             int i2 = i1 % 8;
             int i3 = i1 / 8;
