--- a/net/minecraft/commands/arguments/ObjectiveArgument.java
+++ b/net/minecraft/commands/arguments/ObjectiveArgument.java
@@ -66,7 +66,7 @@
          return SharedSuggestionProvider.suggest(((CommandSourceStack)context.getSource()).getServer().getScoreboard().getObjectiveNames(), builder);
       } else if (context.getSource() instanceof SharedSuggestionProvider) {
          SharedSuggestionProvider sharedsuggestionprovider = (SharedSuggestionProvider)context.getSource();
-         return sharedsuggestionprovider.customSuggestion(context, builder);
+         return sharedsuggestionprovider.customSuggestion((CommandContext<SharedSuggestionProvider>) context, builder);
       } else {
          return Suggestions.empty();
       }
